testsuite/perf_utils.py

```diff
@@ -0,0 +1,102 @@
+        url = url._replace(netloc=url.hostname + f':{url_port}')
+    return url
+
+
+class HyperfoilUtils:
+    """
+        Setup class for hyperfoil test and wrapper of Hyperfoil-python-client.
+    """
+    message_1kb = resources.files('testsuite.resources.performance.files').joinpath('message_1kb.txt')
+
+    def __init__(self, hyperfoil_client, template_filename):
+        self.hyperfoil_client = hyperfoil_client
+        self.factory = HyperfoilFactory(hyperfoil_client)
+        self.benchmark = _load_benchmark(template_filename)
+
+    def create_benchmark(self):
+        """Creates benchmark"""
+        benchmark = self.benchmark.create()
+        return self.factory.benchmark(benchmark).create()
+
+    def update_benchmark(self, benchmark):
+        """Updates benchmark"""
+        self.benchmark.update(benchmark=benchmark)
+
+    def add_shared_template(self, agents_number: int):
+        """Updates benchmark with shared template for hyperfoil agents setup"""
+        agents_dict: dict = {'agents': {}}
+        for i in range(1, agents_number + 1):
+            agent_dict = {'host': 'localhost', 'port': 22, 'stop': True}
+            agents_dict['agents'][f'agent-{i}'] = agent_dict
+        self.benchmark.update(agents_dict)
+
+    def finalizer(self):
+        """Hyperfoil factory opens a lot of file streams, we need to ensure that they are closed."""
+        self.factory.close()
+
+    def add_host(self, url: str, shared_connections: int, **kwargs):
+        """Adds specific url host to the benchmark"""
+        self.benchmark.add_host(url, shared_connections, **kwargs)
+
+    # pylint: disable=consider-using-with
+    def add_file(self, path):
+        """Adds file to the benchmark"""
+        filename = os.path.basename(path)
+        self.factory.file(filename, open(path, 'r', encoding="utf8"))
+
+    def generate_random_file(self, filename: str, size: int):
+        """Generates and adds file with such filename and size to the benchmark"""
+        self.factory.generate_random_file(filename, size)
+
+    def generate_random_files(self, files: dict):
+        """Generates and adds files to the benchmark"""
+        for filename, size in files.items():
+            self.factory.generate_random_file(filename, size)
+
+    def add_user_key_auth(self, user_key, url, filename):
+        """
+        TODO: add method for user key authentication
+        """
+
+    def add_rhsso_auth_token(self, rhsso, client_url, filename):
+        """
+        Adds csv file with data for access token creation. Each row consists of following columns:
+        [authority url, rhsso url, rhsso path, body for token creation]
+        :param rhsso: rhsso service fixture
+        :param client_url: url of desired endpoint to be tested
+        :param filename: name of csv file
+        """
+        rows = []
+        token_url_obj = urlparse(rhsso.well_known['token_endpoint'])
+        token_port = 80 if token_url_obj.scheme == 'http' else 443
```

Comment:
Why are you not using `prepare_url`?

Reply:
good point, it was written before `prepare_url`